DependencyInjection/Compiler/RepositoryAliasPass.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\Common\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\Kernel;
+
+class RepositoryAliasPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasParameter('doctrine.entity_managers')) {
+            return;
+        }
+
+        $entityManagers = $container->getParameter('doctrine.entity_managers');
+        $customRepositories = [];
+
+        foreach ($entityManagers as $name => $serviceName) {
+            $metadataDriverService = sprintf('doctrine.orm.%s_metadata_driver', $name);
+
+            if (!$container->has($metadataDriverService)) {
```

Comment:
It is a private service, but, nota bene, it's a private service from this bundle. 

Using `doctrine.orm.default_entity_manager.metadata_factory` does not seem possible to me. That service depends on the entire entity manager being fully constructed. At the very least, it's not possible with the current extension code structure (because the entity manager is constructed using parent services).